Fix to launch multibranch

michaelneale opened the pull request:
**Decription**
This is a regression fix, a few different things. cc @cliffmeyers @vivek 

Related to https://issues.jenkins-ci.org/browse/JENKINS-36675

**Submitter checklist**
- [x] Change is code complete and matches issue description
- [x] Apppropriate unit or acceptance tests or explaination to why this change has no tests

**Reviewer checklist**
- [x] Run the changes and verified the change matches the issue description
- [x] Reviewed the code

@reviewbybees

vivek commented:
@michaelneale Have not run the PR but the changes looks good to me 🐝

reviewbybees commented:
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

cliffmeyers commented:
Afraid I don't have a chance to run this myself but was just in the code and the changes look sensible to me. RunPipeline is gonna get refactored soon in another ticket and may obviate this change anyway, so good to get the fix in now IMO. :bee

michaelneale commented:
@cliffmeyers @vivek  this is actually 3 fixes:
1) The actual regression that was stopping running even "master" branch
2) I added an encoding call to ensure it can run branches like "feature/blah"
3) I changed the blue message enricher to correctly encode the branch name so that SSE would pick it up correctly.
The 1st one was really a regression

@cliffmeyers we will need another PR/issue for fixing PullRequests.jsx - are you able to open that? I figured you wanted to refactor it anyway to be more in common. It certainly won't cope with names like this the way it is.

michaelneale commented:
@cliffmeyers as I suspected, this doesn't work in folders (obviously neither would it have before). 

I will leave it up to you if you want to merge this in (makes it a bit better, possibly back where it was) or leave it out, for a bigger fix for paths/branches/folders and the ensuing confusion.

michaelneale commented:
@cliffmeyers ok did some snooping, using fullName seems to help, needed to tweak the SSE a bit to ensure freestyle works too in a folder and so on. I am not sure if this helps, but I will leave this call up to you if this moves it forward, or gets in the way of your plans. 

The SSE isn't quite right for things in folders or with "funny" names, but there aren't errors, and you can launch things at least.

michaelneale commented on blueocean-events/src/main/java/io/jenkins/blueocean/events/BlueMessageEnricher.java:
```diff
@@ -38,37 +38,38 @@
         // TODO: Replace once https://issues.jenkins-ci.org/browse/JENKINS-36286 is done
         message.set(EventProps.Jenkins.jenkins_org, OrganizationImpl.INSTANCE.getName());
-        
+
```
this was gulp --fixLint - honest. Scouts honour.

scherler replied:
or your ide, my is configured to remove empty spaces on line endings ;)

michaelneale replied:
I use the best: atom. I do think this was gulp though, as I ran it once to clean things up.

scherler commented:
Have not test it but :bee: for the code

cliffmeyers commented:
Tested this locally and the following scenarios all worked:

# MBP nested in folder, from Branches tab, running both "master" and "feature/XYZ" type branches
# MBP at top-level, same scenarios as above
# Running Pipeline nested in folder, from Activity tab
# Running top-level Pipeline, from Activity tab

Looks good here :bee . I'm merging it now.
